### ISSUE TITLE
Should Wxxxx codes to be marked as failed or still successful? - W3002

### ISSUE THREAD

**Issue by ozbillwang** [aws-cloudformation/cfn-python-lint#1918]
*cfn-lint version: (`cfn-lint --version`)*

tested in 0.44.7 and 0.46.0

*Description of issue.*

The design is 

```
Exit Codes
cfn-lint will return a non zero exit if there are any issues with your template. The value is dependent on the sevirity of the issues found. For each level of discovered error cfn-lint will use bitwise OR to determine the final exit code. This will result in these possibilities.

0 is no issue was found
2 is an error
4 is a warning
6 is an error and a warning
8 is an informational
10 is an error and informational
12 is an warning and informational
14 is an error and a warning and an informational
```

But my question is, 

it is ok to get warning (start with **W means warning, not error, right**?), but don't mark it as failed with exist status code 4

```
$ cfn-lint template.yaml
...
W3002 This code may only work with `package` cli command as the property (Resources/LambdaHelloWorld/Properties/Code) is a string
template.yaml:53:7

$ echo $?
4
```
Could we still get exit status code 0? Because currently I run this command in pipeline. Not 0 exit codes are marked as failed.

and from above codes, exit codes 4, 8 12 could be 0

**Comment by ozbillwang** [comment]
To clarify this problem. I share the Makefile i run with

```
validate: ## Validated CloudFormation Templates...
	docker run -ti --rm -v $(CURDIR):/data -w /data alpine/cfn cfn-lint template.yaml
```

So if I run `make validate`, this build will be marked as failed.

```
$ make validate
...
W3002 This code may only work with `package` cli command as the property (Resources/LambdaHelloWorld/Properties/Code) is a string
template.yaml:53:7

make: *** [Makefile:16: validate] Error 4

echo $?
2
```

Because `make` think it is failed (inside commands returns exit code 4) and reture 2 as error


here is the reference of Makefile return code:

```
2 | There was an error in the makefile.
```
http://www.opussoftware.com/manual/appendices/B1.htm

**Comment by PatMyron** [comment]
Discussed in https://github.com/aws-cloudformation/cfn-python-lint/issues/1586
